src/Databases/MySQL/MaterializedMySQLSyncThread.cpp

```diff
@@ -366,11 +368,14 @@ static inline String rewriteMysqlQueryColumn(mysqlxx::Pool::Entry & connection,
             if (column_type.starts_with("set"))
             {
                 query_columns << (backQuote(column_name) + " + 0");
+                has_set_type = true;
             } else
                 query_columns << backQuote(column_name);
             query_columns << ",";
         }
     }
+    if (!has_set_type)
+        return "*";
```

Comment:
> why this change is needed? looks like should be correct with `query_columns_str` in case `!has_set_type`

query_columns can be empty, checking why...